### DISCUSSION TITLE
How to recreate an GitHub Actions (ubuntu) error?

### DISCUSSION THREAD

**Discussion by Dekker1** [Homebrew/discussions#3366] [Tap maintenance and Homebrew development]
I've recently started a [new tap](https://github.com/Dekker1/homebrew-minizinc) using `brew tap-new`, and I've been presently surprised by how well the CI system works. I've slowly been adding new formulae, but I'm now trying to add one that has got me stumped.

[Mistral](https://github.com/ehebrard/Mistral-2.0) is by no means a straightforward installation, but I managed to write a [formula](https://github.com/Dekker1/homebrew-minizinc/pull/17/files) for it. The test included in the formula works perfectly fine on my computer.

However, submitting the PR, the `ubuntu-latest` test-bot seems to be failing, but gives [little information](https://github.com/Dekker1/homebrew-minizinc/runs/6933263287?check_suite_focus=true). Running only the `macos-latest` test-bot: [it does seem to succeed](https://github.com/Dekker1/homebrew-minizinc/runs/6933070279?check_suite_focus=true).

My main question is thus: **How do I recreate the environment of the `ubuntu-latest` test-bot, to figure out what the problem is?**

I've used the `homebrew/brew` docker image and installed the package there, but the test succeed similar my own computers. Is there a better way to do this?

---
P.S. I must admit that I was initially confused at how the pull request had to be pulled. It was not clear to me that I had to create (and assign) a `pr-pull` label. Instead, I would just use the `brew pr-pull` command from command line which caused some issues with package permissions. I would suggest to have some pointer comments in the workflows.

**Comment by SMillerDev** [comment]
> I would just use the brew pr-pull command from command line which caused some issues with package permissions. I would suggest to have some pointer comments in the workflows.

Pull requests are welcome.

> I've used the homebrew/brew docker image and installed the package there, but the test succeed similar my own computers. Is there a better way to do this?

Could it be because you had an interactive terminal attached when you ran the tests in the container?

**Reply by Dekker1** [reply]
Thank you for the quick reply. I hadn't thought of the interactive element, but after a discussion with one of the software for which I was writing the formula, he is convinced that the problem is the difference in compiler (version). I'm still surprised however that using the `homebrew/brew` docker doesn't reveal the same problem. I would have expected it to behave the same as the linux version used in the GitHub action.

Is there a better way to recreate that environment?

**Reply by SMillerDev** [reply]
No, because the Linux tests for homebrew run in that container.

**Comment by Rylan12** [comment]
> I've used the `homebrew/brew` docker image and installed the package there, but the test succeed similar my own computers. Is there a better way to do this?

One thing to note (not sure that this would explain your issue, though) is that the `homebrew/brew` docker image is built on Ubuntu 22.04 while GitHub's `ubuntu-latest` image is Ubuntu 20.04. You can find out more about what GitHub provides in their `ubuntu-latest` environment [here](https://github.com/actions/virtual-environments#available-environments).
